src/sentry/runner/commands/run.py

```diff
@@ -658,6 +658,18 @@ def handler(signum, frame):
     streamer.run()
 
 
+@run.command("billing-metrics-consumer")
+@log_options()
+@batching_kafka_options("billing-metrics-consumer")
+@configuration
+@click.option("--topic", default="snuba-generic-metrics", help="Topic to get metrics from.")
```

Comment:
You should not need to pass a topic name in the CLI and it is generally a bad practice.
You should not need it because, if you want to override the topic name, `batch_kafka_options` already provides you a [force-topic CLI argument](https://github.com/getsentry/sentry/blob/f94d5af03ed666db8b5ad3898641bdb785ee17e4/src/sentry/runner/commands/run.py#L476-L490).
It is a bad practice because it bypasses the system that maps topics to cluster in the settings. Topics are defined as abstract entities here: https://github.com/getsentry/sentry/blob/master/src/sentry/conf/server.py#L2442-L2472 then mapped to clusters here https://github.com/getsentry/sentry/blob/master/src/sentry/conf/server.py#L2489. The idea is that you could override topic names per environment via settings. That is horribly broken because how settings work in Sentry, still if you expect the topic name to be always passed in, you would be skipping all the logic that maps topic name to cluster which is even worse.

So the right approach is to remove this field and rely on the constant from server.py as a topic name. Also use the force-topic field if, for some reason, in some specific environment there is a reason to override the topic. Should very rarely happen

Reply:
👍 I removed the option.